ui/src/CollectionExerciseDetails.js

```diff
@@ -36,10 +37,14 @@ class CollectionExerciseDetails extends Component {
     newActionRuleType: ''
   }
 
-  componentDidMount() {
+  async componentDidMount() {
     this.getAuthorisedActivities() // Only need to do this once; don't refresh it repeatedly as it changes infrequently
     this.getActionRules()
-    this.getPackCodes()
+
+    await getActionRulePrintTemplates(this.props.surveyId)
```

Comment:
oh because you want to set the state, right? Might have to use a callback function to make this nicer

Reply:
like I said in earlier comment... break this bit out into its own async function, and call it here instead of using `await` in the `componentDidMount` which could cause rendering problems.

Something along the lines of:
```
componentDidMount() {
    this.getAuthorisedActivities()
    this.getActionRules()
    this.getPrintTemplates()
}

getPrintTemplates = async () => {
    const packCodes = await getActionRulePrintTemplates(this.props.surveyId)
    this.setState({ packCodes: packCodes })
}
```